events/wc_undead_events.txt

```diff
@@ -1598,12 +1857,34 @@ character_event = {
 		name = EVTOPTA_WCUND_375
 
 		remove_racial_traits_effect = yes
-		add_trait = creature_abomination
+		
+		# Assign creature trait depending on religion		
+		if = {
+			limit = { religion = throsic } 
+		}
+		add_trait = creature_wicker_golem
+
+		else = {
+			add_trait = creature_abomination
+		}
```

Comment:
Again, it doesn't work because `creature_wicker_golem` is outside the if block.
```suggestion
		if = {
			limit = { religion = throsic }
			add_trait = creature_wicker_golem
		}
		else = {
			add_trait = creature_abomination
		}
```